### PR TITLE
glossary: few terms from intro, wallets, keys chapters

### PR DESCRIPTION
**Pull request by abhiShandy**
I have left terms like "Turing Complete" and "Merkle Patricia Tree" undefined for now

### CODE REVIEW

File: glossary.asciidoc [hunk 1]
```diff
@@ -120,22 +132,42 @@ TODO change for clarity
...
+
 // In case of Ethereum, perhaps, includes a blockchain explorer too?
 IDE (Integrated Development Environment)::
 	An integrated user interface that combines a code editor, compiler, runtime, and a debugger.
 
 Immutable Deployed Code Problem::
   Once a contract's (or library's) code is deployed it becomes immutable. Being able to fix possible bugs and add new features is key for the software development cycle. This represents a challenge for smart contract development.
 
+Inter exchange Client Address Protocol (ICAP)::
+  An Ethereum Address encoding that is partly compatible with the International Bank Account Number (IBAN) encoding, offering a versatile, checksummed and interoperable encoding for Ethereum Addresses. ICAP addresses can encode Ethereum Addresses or common names registered with an Ethereum name registry.
```
Comment (fjrojasgarcia) [inline glossary.asciidoc]: Here I'd add ... ICAP addresses in Ethereum always begin with **XE**. The aim is to introduce a new IBAN country code: XE, Ethereum E prefixed with the "extended" X, as used in non-jurisdictional currencies (e.g. XBT, XRP, XCP).

File: glossary.asciidoc [hunk 2]
```diff
@@ -120,22 +132,42 @@ TODO change for clarity
...
+
 Internal transaction (also "message")::
     A transaction sent from a contract account to another contract account or an EOA.
 
+Key Derivation Function (KDF)::
+  Also known as a password stretching algorithm, it is used by keystore format which to protect against brute-force, dictionary, or rainbow table attacks against the passphrase encryption. It repeatedly hashes the passphrase.
+
+Keystone File::
```
Comment (fjrojasgarcia) [inline glossary.asciidoc]: Sorry but, are you refering to here to Keystore file?

File: glossary.asciidoc [hunk 3]
```diff
@@ -39,6 +39,9 @@ Block::
 Blockchain::
 	A list of blocks validated by the proof-of-work system, each linking to its predecessor all the way to the genesis block. This varies from the Bitcoin protocol's in that it does not have a block limit, it instead uses varying gas limits.
 
+Byzantium::
+  Byzantium is the first of two hard forks for Metropolis development stage.
```
Comment (fjrojasgarcia) [inline glossary.asciidoc]: Perhaps, here we could link it with the Ice Age HF following the **"EIP-649**: Byzantium Difficulty Bomb Delay", which proposes to delay the difficulty bomb for approximately one and a half year.

File: glossary.asciidoc [hunk 4]
```diff
@@ -120,22 +132,42 @@ TODO change for clarity
...
+  Also known as a password stretching algorithm, it is used by keystore format which to protect against brute-force, dictionary, or rainbow table attacks against the passphrase encryption. It repeatedly hashes the passphrase.
+
+Keystone File::
+  A JSON-encoded file that contains a single (randomly generated) private key, encrypted by a passphrase for extra security.
+
 Library::
   A library in Ethereum is a special type of contract that has no payable functions, no fallback function, and no data storage. Therefore, it cannot receive or hold ether, or store data. A library serves as previously deployed code that other contracts can call for read-only computation.
 
+Merkle Patricia Tree::
+
```
Comment (fjrojasgarcia) [inline glossary.asciidoc]: Understood as a place holder, I'd add here a **TODO** tag.

File: glossary.asciidoc [hunk 5]
```diff
@@ -219,14 +257,34 @@ Solidity::
...
+  A decentralised (P2P) storage network. It is used alongwith Web3 and Whisper to build DApps.
+
+Tangerine Whistle::
+  A hard fork at block #2,463,00 to change the gas calculation for certain IO-heavy operations and to clear the accumulated state from a denial of service attack, which exploited the low gas cost of those operations.
+
 Transaction::
-	Data committed to the Ethereum Blockchain signed by an originating account, targeting a specific address. The transaction contains metadata such as the gas limit for the transaction.
+  Data committed to the Ethereum Blockchain signed by an originating account, targeting a specific address. The transaction contains metadata such as the gas limit for the transaction.
+
+Turing Complete::
```
Comment (fjrojasgarcia) [inline glossary.asciidoc]: Perhaps another TODO tag here.

File: glossary.asciidoc (hunk 2 above)
Comment (abhiShandy) [reply glossary.asciidoc]: ya, it's a typo

**Comment by aantonop** [comment]
Nice work. SHIP IT!

**Comment by fjrojasgarcia** [comment]
@abhiShandy  Thanks for your corrections and improvements! Good job!

**Pull request merged**